resources/top_level_overview.markdown

```diff
@@ -0,0 +1,283 @@
+sized blocks, defaulting to 64 MB.  It then stores copies of these blocks across
+nodes in the cluster.  
+
+Using YARN (YARN Applied Resource Negotiator), The resource manager acts as the
+cluster resource manager (no way), as well as the job scheduling facility.  The
+resource manager creates an Application Master daemon (AMD) to monitor the
+life cycle of the job.  One of the initial things that the AMD does is distinguish
+which HDFS blocks are needed for processing.  The AMD requests data from the
+`NameNode` on where the relevant replicated blocks of data are stored. With this
+information, the AMD then requests the resource manager to have map tasks
+process the specified blocks on the same nodes where the data is stored. This
+data locality helps to avoid potentially painful network bottlenecks.
+
+The number of map tasks correspond to the number of created blocks, while the
+number of reducers are configurable via the `mapreduce.jobs.reduces` property.
+Thus for each input split, a map task is spawned.  The number of maps is
+typically driven by the number of HDFS blocks in the input file. This block size
+is configurable, and therefore allows for a workaround when wishing to adjust the
+number of map tasks.
+
+There does exists a `mapred.map.tasks` parameter, however this is simply a hint
+to the `InputFormat` (which describes the input-specification for a Map-Reduce
+job), regarding the number of map tasks.
+
+Thus, e.g., if you expect 10TiB of input data and have 128MiB DFS blocks, you'll
+end up with 81920 maps, unless your `mapred.map.tasks` is even larger.
+Ultimately the `InputFormat` determines the number of maps.
+
+```
+Number of blocks = integer ceiling of input data over block size
+
+= ceil( input_data / block_size )
+= (10 * 2^40) / (2^7 * 2^20)
+= (10 * 2^40) / 2^27
+= 10 * 2^13
+= 10 * 8192
+= 81920
+```
+
+The question remains: What happens when a data record gets split into two
+different blocks?  After all, Hadoop has no idea about what the input data is
+going to look like, so it certainly can't distinguish where one record ends and
+the other begins.
+
+To deal with this, Hadoop uses something called _input splits_.  This is a
+logical representation of that data.  When a job client first calculates the
+input splits, it determines where the first complete record in a block begins,
+and where the last record ends.  In cases where the last record within a block
+has been split off, the input split includes location data on the remnants of
+the record, including the byte offset of the partial record.
+
+
+#### Point 3 & 4: Mapping & Handling Intermediate map Values
+
+For each map task, Hadoop places intermediate values into a circular buffer,
+located in memory, and is configurable under `mapreduce.task.io.sort.mb`
+(defaults to 100MB). This value is the total amount of data that a particular
+map task can output. If a configurable percentage of this limit is met, then the
```

Comment:
instead of "can output", I think it would be better to  say "the total amount of output data that a particular map task can keep in memory.

Reply:
I'm glad you brought this up.  As I was reading through this on the PR I thought this point could be worded much better, and there it is!